src/screens/Onboarding/Onboarding.tsx

```diff
@@ -20,33 +35,307 @@ const ProjectQuery = gql`
+  // TODO: Removed this entirely to solve for the most common case of an existing user with some builds to use as a baseline.
+  // Removing instead of fixing to avoid additional work as this project is past due. We need to revisit this later.
+  const [showInitialBuild, setShowInitialBuild] = useState(showInitialBuildScreen); // !selectedBuild;
+  useEffect(() => {
+    // Watch the value of showInitialBuildScreen, and if it becomes true, set the state to true. This is necessary because Onboarding may render before there is data to determine if there are any builds.
+    if (showInitialBuildScreen) {
+      setShowInitialBuild(true);
+    }
+  }, [showInitialBuildScreen]);
+
+  const [showCatchAChange, setShowCatchAChange] = useState(() => !showInitialBuild);
+  const [initialGitHash, setInitialGitHash] = React.useState(gitInfo.uncommittedHash);
+
+  const onCatchAChange = () => {
+    setInitialGitHash(gitInfo.uncommittedHash);
+    setShowCatchAChange(true);
+  };
+
+  const [runningSecondBuild, setRunningSecondBuild] = React.useState(false);
+
+  // TODO: This design for an error in the Onboarding is incomplete
+  if (localBuildProgress && localBuildProgress.currentStep === "error") {
+    return (
+      <Container>
+        <Stack>
+          <h1>Something went wrong</h1>
+          <p>
+            {Array.isArray(localBuildProgress.originalError)
+              ? localBuildProgress.originalError[0]?.message
+              : localBuildProgress.originalError?.message}
+          </p>
+          <Button small secondary onClick={startDevBuild}>
+            Try again
+          </Button>
+          <Button link onClick={onSkip}>
+            Skip walkthrough
+          </Button>
+        </Stack>
+      </Container>
+    );
+  }
+
+  // TODO: After running a first build, this screen should show the showCatchAChange screen. But if the user restarts Storybook, it comes back to this screen.
+  // Oddly, this seems to be because project.lastBuild is undefined in GraphQL, so at least it doesn't skip out of the onboarding flow.
+  // But we'll need some check to see if one build was run to prompt them to make a change and run another
+  if (showInitialBuild && !localBuildProgress) {
+    return (
+      <Container>
+        <Stack>
+          <div>
+            <VisualTestsIcon />
+            <Heading>Get started with visual testing</Heading>
+            <Text>
+              Take an image snapshot of each story to save their “last known good state” as test
+              baselines.{" "}
+            </Text>
+          </div>
+          <Button small secondary onClick={startDevBuild}>
+            Take snapshots
+          </Button>
+          <Button onClick={onSkip} link>
+            Skip walkthrough
+          </Button>
+        </Stack>
+      </Container>
+    );
+  }
+
+  if (
+    showInitialBuild &&
```

Comment:
This is more of a nit and doesn't have to be addressed in this PR but the logic for what to show could be refactored a bit to make it more readable

Reply:
I will try to improve this in a follow-up. I'm not ready to rewrite the navigation logic yet.

Reply:
I plan to handle more improvements in a follow-up pr, but in the latest changes I have at least simplified the if conditions to make it a little easier to refactor eventually, and removed the no screen case, as there should now be no reasons for it to mistakenly display no screen.